README.md

```diff
@@ -59,16 +59,40 @@ require('satellite').setup {
   handlers = {
     search = {
       enable = true,
+      overlap = true,
+      priority = 10,
+      highlight = {
+        current_match = 'SearchCurrent',
+        other_matches = 'Search',
+      },
     },
     diagnostic = {
       enable = true,
+      overlap = true,
+      priority = 50,
+      highlight = {
```

Comment:
Instead of passing these in via config, I think it would be more scalable to use predefined highlights that have default values.

E.g. for diagnostics we can have:

```lua
vim.api.nvim_set_hl('DiagnosticErrorSV', {link = 'DiagnosticError', default = true})
```

Then users can simply redefine `DiagnosticErrorSV` if they don't like the default.

Reply:
ok I add highlight rules